File: Hero/Models/EndorsementRequest.cs
using System.Collections.Generic;
using WebApiDto.Enum;

namespace Hero.Models
{
    public class EndorsementRequest 
    {
        public int DocumentId { get; set; }

        public int DocumentVersionId { get; set; }

        public int? WordDocumentTemplateId { get; set; }

        public bool IsDraft { get; set; }

        public FileFormat FileFormat { get; set; }

        public IDictionary<string, string> MergeFields { get; set; }

        public IDictionary<string, byte[]> ImageFields { get; set; }

        public IDictionary<string, string> BookmarksToReplace { get; set; }

    }
}

File: Hero/ClientApp/src/app/quote/models/endorsements/EndorsementRequest.ts
import { FileFormat } from "@app/enums";
export class EndorsementRequest {  
    public documentId: number;
    public documentVersionId: number;
    public wordDocumentTemplateId: number;
    public isDraft: boolean;
    public fileFormat: FileFormat;
    public mergeFields: { [key: string]: string; };
    public imageFields: { [key: string]: number[]; };
    public bookmarksToReplace: { [key: string]: string; }; 
}
